operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakRealmImportJob.java

```diff
@@ -0,0 +1,213 @@
+
+import java.util.List;
+import java.util.Optional;
+
+public class KeycloakRealmImportJob extends OperatorManagedResource {
+
+    private final KeycloakRealmImport realmCR;
+    private final Deployment existingDeployment;
+    private final Job existingJob;
+    private final String secretName;
+    private final String volumeName;
+
+    public KeycloakRealmImportJob(KubernetesClient client, KeycloakRealmImport realmCR, String secretName) {
+        super(client, realmCR);
+        this.realmCR = realmCR;
+        this.secretName = secretName;
+        this.volumeName = KubernetesResourceUtil.sanitizeName(secretName + "-volume");
+
+        this.existingJob = fetchExistingJob();
+        this.existingDeployment = fetchExistingDeployment();
+    }
+
+    @Override
+    protected Optional<HasMetadata> getReconciledResource() {
+        if (existingJob == null) {
+            Log.info("Creating a new Job");
+            return Optional.of(createImportJob());
+        } else {
+            Log.info("Job already available");
+            return Optional.empty();
+        }
+    }
+
+    private Job fetchExistingJob() {
+        return client
+                .batch()
+                .v1()
+                .jobs()
+                .inNamespace(getNamespace())
+                .withName(getName())
+                .get();
+    }
+
+    private Deployment fetchExistingDeployment() {
+        return client
+                .apps()
+                .deployments()
+                .inNamespace(getNamespace())
+                .withName(getKeycloakName())
+                .get();
+    }
+
+    private Job buildJob(Container keycloakContainer, Volume secretVolume) {
+        return new JobBuilder()
+                .withNewMetadata()
+                .withName(getName())
+                .withNamespace(getNamespace())
+                .endMetadata()
+                .withNewSpec()
+                .withNewTemplate()
+                .withNewSpec()
+                .withContainers(keycloakContainer)
+                .addToVolumes(secretVolume)
+                .withRestartPolicy("Never")
+                .endSpec()
+                .endTemplate()
+                .endSpec()
+                .build();
+    }
+
+    private Volume buildSecretVolume() {
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withSecret(new SecretVolumeSourceBuilder()
+                        .withSecretName(secretName)
+                        .build())
+                .build();
+    }
+
+    private Job createImportJob() {
+        var keycloakContainer = buildKeycloakJobContainer();
+        var secretVolume = buildSecretVolume();
+        var importJob = buildJob(keycloakContainer, secretVolume);
+
+        return importJob;
+    }
+
+    private Container buildKeycloakJobContainer() {
+        var keycloakContainer =
+            this
+                .existingDeployment
```

Comment:
this `existingDeployment` could be null, right ?

Reply:
Correct, in which case the reconciler will fail and retry.